### ISSUE TITLE
Crash on `VoxelMeshBlock::MESH_UPDATE_NOT_SENT` assertion failure

### ISSUE THREAD

**Issue by JohnDoneth** [Zylann/godot_voxel#291]
Using commit 925aeda.

While using a flat LOD terrain and making edits very quickly with a voxel tool, I hit this error. If I edit the voxels as soon as the game starts, it crashes. If I wait, it works fine until I make lots of edits to the terrain. For example, digging straight down as fast as I can, hits the same assertion failure. It seems like the error is related to crossing over a LOD.

```
ERROR: FATAL: Condition "block->get_mesh_state() != VoxelMeshBlock::MESH_UPDATE_NOT_SENT" is true.
   at: VoxelLodTerrain::_process (modules\voxel\terrain\voxel_lod_terrain.cpp:1502)
```

Sometimes the error is proceeded by some other errors as well regarding the `mesh_block` being null.
```
ERROR: Condition "mesh_block == nullptr" is true. Continuing.
   at: VoxelLodTerrain::flush_pending_lod_edits (modules\voxel\terrain\voxel_lod_terrain.cpp:1782)
ERROR: Condition "mesh_block == nullptr" is true. Continuing.
   at: VoxelLodTerrain::flush_pending_lod_edits (modules\voxel\terrain\voxel_lod_terrain.cpp:1782)
ERROR: FATAL: Condition "block->get_mesh_state() != VoxelMeshBlock::MESH_UPDATE_NOT_SENT" is true.
   at: VoxelLodTerrain::_process (modules\voxel\terrain\voxel_lod_terrain.cpp:1502)
```

**Comment by Zylann** [comment]
I can get an earlier error when doing this
```
ERROR: VoxelLodTerrain::flush_pending_lod_edits: Condition "mesh_block == nullptr" is true. Continuing.
   At: modules\voxel\terrain\voxel_lod_terrain.cpp:1782
```
Trying to edit anywhere while blocks are still loading is not handled that well at the moment, I need some time to improve this

**Comment by Zylann** [comment]
I fixed a few things in 144affe42d444c78821eae2e2b01e11b97cfc6b0
Let me know if it fixes your issue

**Comment by JohnDoneth** [comment]
Works great! Sometimes this assertion error comes up but the crashing is gone. Thanks!

```
ERROR: Condition "block == nullptr" is true.
   at: VoxelLodTerrain::post_edit_block_lod0 (modules\voxel\terrain\voxel_lod_terrain.cpp:434)
```
